### PR TITLE
Pull in v4 features

### LINKED ISSUE #11: MatchComms Support
Would be nice to include matchcomms / tmcp support into the core. This would have the added benefit of making it team-specific so that each team only sees its own messages.

### PR DESCRIPTION
**Pull request by VirxEC**
- Psyonix bots now have their names and loadouts
- State setting
- Match comms + quickchat (closes #11)
- Better logging
- Fix bug in `GameTickPacket` 
- Close old bot processes when match ends or when a new one starts
- Proper support for Continue and Spawn

### CODE REVIEW

File: RLBotCS/ManagerTools/Logging.cs [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+
+        var logLevelColors = GetLogLevelColors(logLevel);
+        var logLevelString = GetLogLevelString(logLevel);
+        var message = formatter(state, exception);
+
+        lock (_lock)
+        {
+            var logBuilder = new StringBuilder();
+            logBuilder.Append(
+                $"{logLevelColors[0]}{DateTime.Now:yyyy-MM-dd HH:mm:ss}{Reset} "
```
Comment (NicEastvillage) [inline RLBotCS/ManagerTools/Logging.cs:53]: Including the date is a bit excessive. Let's save some space
```suggestion
                $"{logLevelColors[0]}{DateTime.Now:HH:mm:ss}{Reset} "
```

File: RLBotCS/ManagerTools/Logging.cs [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+    private static readonly LogLevel LoggingLevel = LogLevel.Information;
+    private static readonly string AppName = "RLBotServer".PadLeft(12);
+
+    private readonly string _name;
+    private readonly LogLevel _minLevel;
+    private static readonly object _lock = new object();
+
+    public Logging(string name, LogLevel minLevel)
+    {
+        _name = name.PadLeft(20);
```
Comment (NicEastvillage) [inline RLBotCS/ManagerTools/Logging.cs:25]:
```suggestion
        _name = name.PadLeft(16);
```

File: RLBotCS/ManagerTools/QuickChat.cs [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+                {
+                    Text = chat.Item2.Display,
+                    Y = 10f / 1920f,
+                    X = xVal / 1080f,
+                    Scale = 1,
+                    Foreground = textColor,
+                    Background = BackgroundColor,
+                    HAlign = TextHAlign.Left,
+                    VAlign = TextVAlign.Top,
+                };
```
Comment (NicEastvillage) [inline RLBotCS/ManagerTools/QuickChat.cs:90]: Have you tested that it renders correctly on multiple resolutions?

**Review by NicEastvillage — APPROVED** [review]
Some comments and suggestions. Otherwise approved.

File: RLBotCS/ManagerTools/QuickChat.cs (hunk 3 above)
Comment (VirxEC) [reply RLBotCS/ManagerTools/QuickChat.cs:90]: This behavior was taken directly from RLBot v4

**Pull request merged**